### ISSUE TITLE
ZMQ Network Down Detection / Dropped Recovery

### ISSUE THREAD

**Issue by zangjiucheng** [waterloo-rocketry/omnibus#302]
- Current our omnibus server rely on the QMZ network, since it can cause any issue went down. We hope to have a detector of network change and when that happen try to reconnect it after a timeout. 
- Currently only solution for us is manual restart 
- (If the auto-detect feature is not available, set up a manual button to try to reconnect to the server.)

**Comment by ChrisYx511** [comment]
Mildly late, but testing behaviour this week 🫡🫡

**Comment by zangjiucheng** [comment]
Thanks, I will go more detail about this issue in following weeks.

**Comment by ChrisYx511** [comment]
Testing happened, seems to always recover on my end. I will have to go validate on actual hardware in the bay this week. 
I thin regardless whether we successfully detect offline or not, we should always just log on the DAQ computer going forward.

**Comment by zangjiucheng** [comment]
Yes, sorry I didn't update on this. I believe I already test on disconnect with the server on my side (MacOS). Don't know if that's windows only behaviour. Or Ni Source block process of Server.

**Comment by ChrisYx511** [comment]
Also, offline detection can definitely be implemented simply by listening to the Omnibus broadcast IP packets.
